process/block/shardblock.go

```diff
@@ -1912,6 +1912,12 @@ func (sp *shardProcessor) createMiniBlocks(haveTime func() bool, randomness []by
 	if sp.flagScheduledMiniBlocks.IsSet() {
 		miniBlocks = sp.scheduledTxsExecutionHandler.GetScheduledMiniBlocks()
 		sp.txCoordinator.AddTxsFromMiniBlocks(miniBlocks)
+
+		scheduledTxs := sp.scheduledTxsExecutionHandler.GetScheduledIntermediateTxs()
+		invalidTxs := scheduledTxs[block.InvalidBlock]
```

Comment:
As how the func (txs *transactions) AddTransactions is written right now, you can directly replace the L1917-L1920 with one line:  sp.txCoordinator.AddTransactions(scheduledTxs[block.InvalidBlock], block.TxBlock)

Reply:
done